scripts/dev/infra-sync.sh

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env bash
+# Fetches cluster artifacts from infra.rox.systems and merged them into a kubeconfig file.
+
+set -euo pipefail
+
+os=$(uname -o)
+infra_dir="${HOME}/.kube/infra"
+clusters_dir="${infra_dir}/clusters"
+infra_kubeconfig="${INFRA_KUBECONFIG:-${infra_dir}/kubeconfig}"
+extra_kubeconfigs="${EXTRA_KUBECONFIGS:-}"
+
+delete_cluster_artifacts() {
```

Comment:
This is deleting previous cached artifacts, correct?
Two things come to mind: 
1. How about renaming this to indicate we are clearing a cache? Something like `clean_infra_cluster_cache`? (Feels a bit long, though)
2. Personally, I would run this quite often. I know that downloading artifacts takes bit of time. Would it be sensible to check whether we can/should keep artifacts of clusters that are still live?

Reply:
Ad 1: I will touch the code again to incorporate your feedback.
Ad 2: I don't want to do that as part of this PR. That's just an optimization -- a reasonable one, of course. But it wasn't my intention here to extend the scope in such a way.